async.js

```diff
@@ -0,0 +1,19 @@
+let XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest;
+
+const request = new XMLHttpRequest();
+request.open('GET', 'https://pokeapi.co/api/v2/pokemon/ditto');
```

Comment:
а что если я хочу запросить другого покемона? Можем мы как то улучшить наш код в этом месте? Что думаешь?

Reply:
function getPokemon(pokType) {
  const request = new XMLHttpRequest();
  request.open('GET', `https://pokeapi.co/api/v2/pokemon/${pokType}`);
  request.send();

  request.addEventListener('load', function () {
    let { abilities } = JSON.parse(this.responseText);
    const { url } = abilities[0].ability;

    const request = new XMLHttpRequest();
    request.open('GET', url);
    request.send();

    request.addEventListener('load', function () {
      let result = JSON.parse(this.responseText);
      console.log(result.effect_entries[1]);
    });
  });
}

Reply:
`function getPokemon(pokType) {
  const request = new XMLHttpRequest();
  request.open('GET', `https://pokeapi.co/api/v2/pokemon/${pokType}`);
  request.send();

  request.addEventListener('load', function () {
    let { abilities } = JSON.parse(this.responseText);
    const { url } = abilities[0].ability;

    const request = new XMLHttpRequest();
    request.open('GET', url);
    request.send();

    request.addEventListener('load', function () {
      let result = JSON.parse(this.responseText);
      console.log(result.effect_entries[1]);
    });
  });
}`

Reply:
```suggestion
function getPokemon(pokType) {
  const request = new XMLHttpRequest();
  request.open('GET', `https://pokeapi.co/api/v2/pokemon/${pokType}`);
  request.send();

  request.addEventListener('load', function () {
    let { abilities } = JSON.parse(this.responseText);
    const { url } = abilities[0].ability;

    const request = new XMLHttpRequest();
    request.open('GET', url);
    request.send();

    request.addEventListener('load', function () {
      let result = JSON.parse(this.responseText);
      console.log(result.effect_entries[1]);
    });
  });
}
```

Reply:
Не знаю точно, как обрабатывать ошибки используя xmlhttprequest библиотеку в node js - хотелось бы пояснений
```suggestion
function getPokemon(pokType) {
  // try {
  const request = new XMLHttpRequest();

  request.open('GET', `https://pokeapi.co/api/v2/pokemon/${pokType}`);

  request.send();
  request.addEventListener('load', function () {
    let { abilities } = JSON.parse(this.responseText);
    const { url } = abilities[0].ability;

    const request = new XMLHttpRequest();
    request.open('GET', url);
    request.send();

    request.addEventListener('load', function () {
      let result = JSON.parse(this.responseText);
      console.log(result.effect_entries[1]);
    });
  });
  request.onload = function () {
    if (this.status >= 400) {
      console.error(`Server status: ${this.status} Server response: ${this.responseText}`);
    }
  };
}
```

Reply:
А что говорит документация?

Reply:
Есть вроде подобный метод, но я  не могу понять как он рабоатет
request.onerror = function () {
    
  };

Reply:
Ну так точно так же как и другие, у тебя если будет ошибка , то отработает колбек который подписан на событие ошибки. Разницы особой нет как подписку на ошибку сделать.